crates/noirc_frontend/src/parser/errors.rs

```diff
@@ -126,11 +126,17 @@ impl chumsky::Error<Token> for ParserError {
         self.expected_tokens.append(&mut other.expected_tokens);
         self.expected_labels.append(&mut other.expected_labels);
 
+        if self.reason.as_deref() != Some("recovery failure")
+            && other.reason.as_deref() != Some("recovery failure")
+        {
+            // We compare spans, except for errors for recovery, because the span for recovery does not match
+            assert_eq!(self.span, other.span);
+        }
+
```

Comment:
This is a workaround: since spans of the recovery failure alternative do not match the spans of other alternatives, only check that spans match if neither error comes from a recovery failure.

Reply:
I am not sure whether we should drop recovery failures during this merging. They are meant to never be shown to the user.

Reply:
They should likely be dropped just in case, although I'm not aware of any time in which we ever merge a recovery error with a normal user-facing error.